### PR TITLE
Setup react testing library

### PR DESCRIPTION
**Pull request by adamjarling**
@Dananji Feel free to pull this down and mess with it, see what you think as a starting point?

### CODE REVIEW

**Comment by Dananji** [comment]
@adamjarling This looks great! I feel the testing code is more readable with this library unlike Enzyme.

File: src/components/__tests__/HeadingForm.test.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+test('heading title input shows proper validation messages', async () => {
+  const { container, getByLabelText, getByTestId, debug } = renderWithRedux(
+    <HeadingForm />,
+    {
+      initialState
+    }
+  );
+
+  const titleInput = getByLabelText(/title/i);
```
Comment (Dananji) [inline src/components/__tests__/HeadingForm.test.js]: @adamjarling Question: here the `getByLabelText(/title/i)` => is this something like Regex?

Comment (adamjarling) [reply src/components/__tests__/HeadingForm.test.js]: Yea, case insensitive regex.   They use is all over the place in the docs.

**Comment by adamjarling** [comment]
@Dananji Yea, I think the basic premise is to NOT test implementation details.  We don't necessarily need to go crazy mocking all kinds of state and props and other stuff, as Enzyme can easily encourage. 

Another line I read which really hit home for me was, for example if a refactor of some code breaks a test, where the UI and input/output remains the same, then the test is probably tied to the implementation details and the test could probably be simplified.  Refactoring code should not break tests, unless the component itself changes.

The documents are pretty good for the package.  I imagine we'll learn a lot along the way.

**Comment by Dananji** [comment]
@adamjarling I think the starting point for this would be establishing test a data-set covering all possible test cases (getting an idea from the previous test-suite), ensuring that the structure and Peaks data is consistent in this (all the timespans in structure is represented in a segment in the mocked Peaks instance). 

And next 2 steps would be
1. Test for the basic validation logic in both `StructuralMetadaUtils` and `WaveformDataUtils` modules
2. Test each component, to ensure they function as intended

**Comment by adamjarling** [comment]
Yea, sounds good.  Let's chat about it tomorrow afternoon maybe for a bit?   I'm in an interview rest of day today, and in a workshop until 2pm CT tomorrow.

**Pull request merged**